src/Endpoints/Delegates/HandlesDocuments.php

```diff
@@ -22,10 +22,20 @@ public function getDocument($documentId, array $fields = null)
 
     public function getDocuments(DocumentsQuery $options = null): DocumentsResults
     {
-        $query = isset($options) ? $options->toArray() : [];
-        $response = $this->http->get(self::PATH.'/'.$this->uid.'/documents', $query);
+        try {
+            $options = $options ?? new DocumentsQuery();
+            $query = $options->toArray();
+
+            if ($options->hasFilter()) {
+                $response = $this->http->post(self::PATH.'/'.$this->uid.'/documents/fetch', $query);
+            } else {
+                $response = $this->http->get(self::PATH.'/'.$this->uid.'/documents', $query);
+            }
 
-        return new DocumentsResults($response);
+            return new DocumentsResults($response);
+        } catch (\Exception $e) {
```

Comment:
this should probably be a more specialized exception from http than a generic one

Reply:
Yes, I agree with you! I'm awaiting an issue that @alallema will create in the integration-guides. We will have to check it across every SDK. 
For example, we don't have a "BaseMeilisearchException" class that we can use across the package to check if anything happens. Today all of our exceptions are extending Exception 😢.

Reply:
It's out:
- [About the error handler](https://github.com/meilisearch/integration-guides/issues/267)
- [About the hint](https://github.com/meilisearch/integration-guides/issues/268)